### PR TITLE
Simplify short methods in `BaseCoordinateFrame`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

I had a look at some of the `BaseCoordinateFrame` methods, in particular the ones that are so short that the Ruff rules such as [PLR0912 (too-many-branches)](https://docs.astral.sh/ruff/rules/too-many-branches) or [PLR0915 (too-many-statements)](https://docs.astral.sh/ruff/rules/too-many-statements) would not highlight them.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
Nice! All are better and most are more readable too.

**Comment by mhvk** [comment]
I think `readthedocs` has problems - I saw the same immediate failure elsewhere. ping @pllim (sorry!)

File: astropy/coordinates/baseframe.py [hunk 1]
```diff
@@ -867,28 +864,17 @@ def representation_info(self):
...
-            return out
+            return {}
         repr_attrs = self.representation_info[repr_or_diff_cls]
-        repr_names = repr_attrs["names"]
-        repr_units = repr_attrs["units"]
-        for repr_name, repr_unit in zip(repr_names, repr_units):
-            if repr_unit:
-                out[repr_name] = repr_unit
-        return out
+        return {k: v for k, v in zip(repr_attrs["names"], repr_attrs["units"]) if v}
```
Comment (nstarman) [inline astropy/coordinates/baseframe.py:877]:
```suggestion
        return {k: v for k, v in zip(repr_attrs["names"], repr_attrs["units"], strict=True) if v}
```

Comment (nstarman) [reply astropy/coordinates/baseframe.py:877]: Py 3.10+ (https://peps.python.org/pep-0618/)

File: astropy/coordinates/baseframe.py [hunk 2]
```diff
@@ -1857,27 +1831,21 @@ def separation_3d(self, other):
             )
 
         # do this first just in case the conversion somehow creates a distance
-        other_in_self_system = getattr(other, "frame", other).transform_to(self)
+        other = getattr(other, "frame", other).transform_to(self)
 
-        if issubclass(other_in_self_system.__class__, r.UnitSphericalRepresentation):
+        if issubclass(type(other), r.UnitSphericalRepresentation):
```
Comment (nstarman) [inline astropy/coordinates/baseframe.py:1836]: is this better than 
```suggestion
        if isinstance(other, r.UnitSphericalRepresentation):
```

**Comment by pllim** [comment]
Re: https://github.com/astropy/astropy/pull/16509#issuecomment-2135662625

Since there are suggested changes, might as well wait for the next commit and see. 🤞

File: astropy/coordinates/baseframe.py (hunk 1 above)
Comment (eerovaher) [reply astropy/coordinates/baseframe.py:877]: There are situations where the `strict` argument is very useful, but explicitly specifying it everywhere sounds too verbose to me. I can't tell if using it here is helpful or not.

Comment (mhvk) [reply astropy/coordinates/baseframe.py:877]: I think it is very unlikely this would ever fail, as these are set at class definition time.

Comment (neutrinoceros) [reply astropy/coordinates/baseframe.py:877]: > these are set at class definition time.

I guess your point can be interpreted as an argument both ways, but to me it sounds more like `strict=True` is correct (hence desirable) here.

Comment (mhvk) [reply astropy/coordinates/baseframe.py:877]: I think class definition might ensure this is right, but didn't check - to me, it seems overkill.

Comment (eerovaher) [reply astropy/coordinates/baseframe.py:877]: I'm inclined to agree with @mhvk that the verbosity of having to specify `strict=True` here is not worth it.

**Review by mhvk — APPROVED** [review]
Definitely better with the `isinstance`. Reapproving.

**Pull request merged**